### ISSUE TITLE
Single-axis window scatter has same coveredDims overlap pattern as multi-axis and should reuse shared code

### ISSUE THREAD

**Issue by tillahoffmann** [tillahoffmann/jax-mps#164]
## Context

PR #161 fixed `HandleScatter` for the multi-axis general-point path. The root cause was that the `newShape` reshape loop was partitioning operand dims into

```
coveredDims = inserted ∪ batching ∪ scatter_dims_to_operand_dims
```

and emitting a size-1 entry for every covered dim. This is wrong for dims that are in `scatter_dims_to_operand_dims` but **not** in `inserted_window_dims ∪ input_batching_dims` — those *do* have a real (often size-1) slot in the transposed-updates shape and must consume `windowIdx`. The fix uses the spec-exact predicate `insertedSet ∪ batchingSet`.

## Follow-up

The single-axis window scatter branch in `src/pjrt_plugin/ops/gather_scatter.cc` (around line 580 after #161 lands) constructs `coveredDims` with the same overlap pattern:

```cpp
std::set<int> coveredDims;
coveredDims.insert(scatterDim);
for (auto d : insertedWindowDims)
    coveredDims.insert(static_cast<int>(d));
for (auto d : inputBatchingDims)
    coveredDims.insert(static_cast<int>(d));
...
if (coveredDims.count(d)) {
    // emit size 1
}
```

By the same reasoning, `scatterDim` should not be unconditionally collapsed: if it is also in `insertedWindowDims` then its `tShape` slot is absent and a size-1 entry is correct; otherwise `tShape` has a real slot for it that needs to be consumed.

In practice the single-axis path is reached when `scatterDimsToOperandDims.size() == 1`, and a repro that exercises the bug would have a single scatter axis with window-extent > 1 plus mixed inserted dims that don't include that scatter axis. I did not construct such a case while working on #161, so the bug is **suspected, not confirmed**.

## Suggested action

1. Try to construct a `jnp.array(...).at[...]` expression that lowers to single-axis window scatter with `scatterDim ∉ insertedWindowDims` and `insertedWindowDims` non-empty.
2. If reproducible, apply the same `insertedSet ∪ batchingSet` discriminator in the single-axis branch.
3. Add a regression test to `tests/configs/slice.py`.

Referenced from PR #161.
